solidity/contracts/converter/BancorConverterRegistry.sol

```diff
@@ -76,14 +78,113 @@ contract BancorConverterRegistry is IBancorConverterRegistry, ContractRegistryCl
     }
 
     /**
-      * @dev adds a converter to the registry
-      * anyone can add a converter to the registry, as long as the converter is active and valid
+      * @dev creates a liquid-token converter and adds it to the registry
+      * 
+      * @param _smartTokenName token name
+      * @param _smartTokenSymbol token symbol
+      * @param _smartTokenDecimals token decimals
+      * @param _maxConversionFee maximum conversion-fee
+      * @param _reserveToken reserve token
+      * @param _reserveRatio reserve ratio
+      * @param _reserveAmount reserve amount
+    */
+    function newLiquidToken(
+        string _smartTokenName,
+        string _smartTokenSymbol,
+        uint8 _smartTokenDecimals,
+        uint32 _maxConversionFee,
+        IERC20Token _reserveToken,
+        uint32 _reserveRatio,
+        uint256 _reserveAmount
+    )
+    public payable
+    {
+        IBancorConverterFactory factory = IBancorConverterFactory(addressOf(BANCOR_CONVERTER_FACTORY));
+        SmartToken token = new SmartToken(_smartTokenName, _smartTokenSymbol, _smartTokenDecimals);
+        IBancorConverter converter = IBancorConverter(factory.createConverter(token, registry, _maxConversionFee, IERC20Token(0), 0));
+
+        converter.acceptOwnership();
+
+        if (_reserveToken != address(0)) {
+            require(msg.value == 0);
+            converter.addReserve(_reserveToken, _reserveRatio);

```

Comment:
I believe we can actually move it to the constructor now that ether has a non zero address.
But on the other hand I'm thinking of getting rid of the 2 extra ctor arguments (reserve/weight) once this function is the main function for creating a new converter

Reply:
@yudilevi : We most certainly need to adjust this code to the new eth-reserve address.

Reply:
@yudilevi : please see updated code.